src/Microsoft.OpenApi/Models/OpenApiError.cs

```diff
@@ -3,20 +3,18 @@
 
 using Microsoft.OpenApi.Exceptions;
 
-namespace Microsoft.OpenApi.Readers
+namespace Microsoft.OpenApi.Models
 {
     /// <summary>
-    /// Error related to reading Open API YAML/JSON.
+    /// Error related to the Open API Document.
     /// </summary>
     public class OpenApiError
```

Comment:
OpenApiError should be abstract, so that it can't be instantiated.

Reply:
@PerthCharern  I have to move OpenApiReferenceResolver in an upcoming PR to the OpenAPI Model project and it raises errors.  I was considering just creating a OpenApiError for that, but maybe I will create an OpenApiReferenceError.  However, if we make OpenApiError abstract I have to consider whether it should just be an interface.... decisions, decisions.

Reply:
Making this abstract signals that each processor should have their own error classes, and it's easy to filter out errors of the same type from the OpenApiDiagnostic. 

I would argue against making this an interface. The common constructor that you have now makes sense - it can be leveraged by children classes.

---
In reply to: [179584501](https://github.com/Microsoft/OpenAPI.NET/pull/235#discussion_r179584501) [](ancestors = 179584501)